app/src/main/java/de/tum/in/tumcampusapp/activities/BarrierFreeFacilitiesActivity.java

```diff
@@ -0,0 +1,143 @@
+import de.tum.in.tumcampusapp.auxiliary.NetUtils;
+import de.tum.in.tumcampusapp.auxiliary.Utils;
+import de.tum.in.tumcampusapp.managers.LocationManager;
+import de.tum.in.tumcampusapp.managers.RecentsManager;
+import de.tum.in.tumcampusapp.models.tumcabe.RoomFinderRoom;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import se.emilsjolander.stickylistheaders.StickyListHeadersListView;
+
+public class BarrierFreeFacilitiesActivity extends ActivityForLoadingInBackground<Void, Optional<List<RoomFinderRoom>>>
+        implements AdapterView.OnItemSelectedListener, AdapterView.OnItemClickListener{
+    private int selectedFacilityPage = -1;
+    private RoomFinderListAdapter adapter;
+    private StickyListHeadersListView stickyList;
+    private List<RoomFinderRoom> facilities;
+
+    private RecentsManager recentsManager;
+    private LocationManager locationManager;
+
+    public BarrierFreeFacilitiesActivity(){
+        super(R.layout.activity_barrier_free_facilities);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Activate action bar
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+            getSupportActionBar().setHomeButtonEnabled(true);
+        }
+
+        // set spinner
+        Spinner spinner = (Spinner) findViewById(R.id.spinnerToolbar);
+        spinner.setOnItemSelectedListener(this);
+
+        // manager
+        recentsManager = new RecentsManager(this, RecentsManager.ROOMS);
+        locationManager = new LocationManager(this);
+
+        stickyList = (StickyListHeadersListView) findViewById(R.id.activity_barrier_free_facilities_list_view);
+    }
+
+    @Override
+    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        selectedFacilityPage = position;
+        startLoading();
+    }
+
+    @Override
+    protected Optional<List<RoomFinderRoom>> onLoadInBackground(Void... arg) {
+        showLoadingStart();
+        List<RoomFinderRoom> result = null;
+        TUMCabeClient cabeClient = TUMCabeClient.getInstance(this);
+
+        try {
+            switch (selectedFacilityPage) {
+                case 0:
+                    // Nearby staff - need to fetch building id first
+                    Optional<String> buildingId = locationManager.getBuildingIDFromCurrentLocation();
+                    if(buildingId.isPresent()){
+                        result = cabeClient.getListOfNearbyFacilities(buildingId.get());
+                    } else {
+                        return Optional.absent();
+                    }
+                    break;
+
+                case 1:
+                    result = cabeClient.getListOfToilets();
+                    break;
+
+                case 2:
+                    result = cabeClient.getListOfElevators();
+                    break;
+            }
```

Comment:
selectedFacilityPage is initialized to -1
However, I don't see handling of this case? How about a default switch label?